### PR TITLE
Fix image path in efficient_ai.qmd and update data_engineering.qmd

### PR DESCRIPTION
**Pull request by Sara-Khosravi**
Before submitting your Pull Request, please ensure that you have carefully reviewed and completed all items on this checklist.

1. **Content**
    - [ ] The chapter content is complete and covers the topic in detail.
    - [ ] All technical terms are well-defined and explained.
    - [ ] Any code snippets or algorithms are well-documented and tested.
    - [ ] The chapter follows a logical flow and structure.
  
2. **References & Citations**
    - [ ] All references are correctly listed at the end of the chapter.
    - [ ] In-text citations are used appropriately and match the references.
    - [ ] All figures, tables, and images have proper sources and are cited correctly.
  
3. **Quarto Website Rendering**
    - [ ] The chapter has been locally built and tested using Quarto.
    - [ ] All images, figures, and tables render properly without any glitches.
    - [ ] All images have a source or they are properly linked to external sites. 
    - [ ] Any interactive elements or widgets work as intended.
    - [ ] The chapter's formatting is consistent with the rest of the book.
  
4. **Grammar & Style**
    - [ ] The chapter has been proofread for grammar and spelling errors.
    - [ ] The writing style is consistent with the rest of the book.
    - [ ] Any jargon is clearly explained or avoided where possible.
  
5. **Collaboration**
    - [ ] All group members have reviewed and approved the chapter.
    - [ ] Any feedback from previous reviews or discussions has been addressed.
  
6. **Miscellaneous**
    - [ ] All external links (if any) are working and lead to the intended destinations.
    - [ ] If datasets or external resources are used, they are properly credited and linked.
    - [ ] Any necessary permissions for reused content have been obtained.
  
7. **Final Steps**
    - [ ] The chapter is pushed to the correct branch on the repository.
    - [ ] The Pull Request is made with a clear title and description.
    - [ ] The Pull Request includes any necessary labels or tags.
    - [ ] The Pull Request mentions any stakeholders or reviewers who should take a look.

### CODE REVIEW

**Comment by Sara-Khosravi** [comment]
**Hi Professor Vijay,**

I hope all is well! I've made significant updates and fixes that are crucial for our project's progress:

I fixed the image path in efficient_ai: I corrected the reference to ensure it is rendered correctly in the document.
Updated data_engineering: I refined the content for better clarity and coherence.
I used Quarto for rendering to ensure the changes were adequately reflected in the output.

Additionally, I am working on telecom outage prediction and am fully committed to applying TinyML in this domain to the best of my abilities.

Working on this book has been a truly enjoyable experience, especially given my over 7 years of industry experience. I am deeply committed to this project and look forward to our continued collaboration.

Please let me know your feedback, and I am ready to work on other chapters.

Warm regards,
Sara

**Comment by profvjreddi** [comment]
@Sara-Khosravi thanks again for these edits. In the future, could you please make sure that you modify only one file at a time, as it is easier to do merges and rollbacks if needed?

**Comment by Sara-Khosravi** [comment]
Hi Vijay,

Thank you for your positive feedback. I am glad that you are willing to
take a look at my project.

Regarding your question, I apologize for any oversight. I will double-check
to ensure I used the latest version of the repository. If it turns out I
didn't, I will make sure to synchronize with the latest version and update
my changes accordingly.

Thank you for bringing this to my attention. I appreciate your
understanding and patience.

Best regards,
Sara Khosravi

File: contents/data_engineering/data_engineering.qmd [hunk 1]
```diff
@@ -459,24 +510,35 @@ New data regulations also impact licensing practices. The legislative landscape
...
...llection. Data can be sourced from diverse means, including existing datasets, web scraping, crowdsourcing, and synthetic data generation. Each approach involves tradeoffs between cost, speed, privacy, and specificity. Once data is collected, thoughtful labeling through manual or AI-assisted annotation enables the creation of high-quality training datasets. Proper storage in databases, warehouses, or lakes facilitates easy access and analysis. Metadata provides contextual details about the data. Data processing transforms raw data into a clean, consistent format for machine learning model development. Throughout this pipeline, transparency through documentation and provenance tracking is crucial for ethics, auditability, and reproducibility. Data licensing protocols also govern legal data access and use. Key challenges in data engineering include privacy risks, representation gaps, legal restrictions around proprietary data, and the need to balance competing constraints like speed versus quality. By thoughtfully engineering high-quality training data, machine learning practitioners can develop accurate, robust, and responsible AI systems, including embedded and TinyML applications.
```
Comment (profvjreddi) [inline contents/data_engineering/data_engineering.qmd:537]: Does this need to be deleted based on the above text? 

There seems to be some repetition.

**Review by profvjreddi — CHANGES_REQUESTED** [review]
Could you please look over the comments and make the small tweaks please?

**Comment by profvjreddi** [comment]
Cool. Thanks for taking a look. I left some comments.


Also, in the future, do you think we could make small changes because that
makes it easier to merge things in.

Vijay Janapa Reddi, Ph. D. |
John L. Loeb Associate Professor of Engineering and Applied Sciences |
John A. Paulson School of Engineering and Applied Sciences |
Science and Engineering Complex (SEC) | 150 Western Ave, Room #5.305 |
Boston, MA 02134 |
Harvard University | My Website
<http://scholar.harvard.edu/vijay-janapa-reddi> | Google Scholar
<https://scholar.google.com/citations?hl=en&user=gy4UVGcAAAAJ&view_op=list_works&sortby=pubdate>
 | Edge Computing Lab <https://edge.seas.harvard.edu> | Book Meeting
<https://fantastical.app/vjreddi/> | Contact Admin
<https://scholar.harvard.edu/vijay-janapa-reddi/contact> |

**Comment by Sara-Khosravi** [comment]
It is greatly appreciated, Vijay. I will look at it and fix it as soon as
possible. Also, I am working on the next section. I will update you after I
consider your comment.

Have a wonderful time.

**Comment by profvjreddi** [comment]
Looked over this and the changes are already merged in from other edits we did, so these updates are already in!

**Pull request closed without merging**